discovery/service.go

```diff
@@ -714,11 +721,18 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []l
 		// announcement. If we didn't receive the opposite half of the
 		// proof than we should store it this one, and wait for
 		// opposite to be received.
-		oppositeKey := newProofKey(chanInfo.ChannelID, !nMsg.isRemote)
-		oppositeProof, ok := d.waitingProofs[oppositeKey]
+		proof := newWaitingProof(nMsg.isRemote, msg)
+		oppositeKey := proof.oppositeKey()
+		oppositeProofIndex, ok := d.proofToIndex[oppositeKey]
 		if !ok {
-			key := newProofKey(chanInfo.ChannelID, nMsg.isRemote)
-			d.waitingProofs[key] = msg
+			index, err := d.proofStore.Add(proof)
+			if err != nil {
+				nMsg.err <- errors.Errorf("unable to store "+
+					"the proof for short_chan_id=%v: %v",
+					shortChanID, err)
+				return nil
```

Comment:
I think an error should be returned her as well, we we're able to add the proof to our persistent storage, therefore this'll potentially cause us to mishandle any of the follow up messages are part of this channel announcement.